sdk/database/helper/dbutil/dbutil.go

```diff
@@ -18,13 +18,24 @@ var (
 	ErrEmptyRotationStatement = errors.New("empty rotation statements")
 )
 
-// Query templates a query for us.
-func QueryHelper(tpl string, data map[string]string) string {
-	for k, v := range data {
-		tpl = strings.ReplaceAll(tpl, fmt.Sprintf("{{%s}}", k), v)
-	}
+var queryHelperRegex = regexp.MustCompile(`{{[^{}]+}}`)
```

Comment:
I don't have the original context so I won't do a full review, but if we do go with a regex I think we should be extremely prescriptive with it to more precisely match what we're looking for, e.g. here's a little Go playground test case for a key that has a `}` in it (which probably isn't valid for other reasons, but it's just to illustrate the point) https://go.dev/play/p/qUXqc397xkY.

So I'd rather see a regex that specifically looks for the keys we know how to replace, e.g. `fmt.Sprintf(`{{(%s)}}`, strings.Join(keys, "|"))`.

Reply:
The major point of this change is to do only *one* pass over the input string, not one per replaceable parameter, so the regex needs to stay as is.

However, I recognise I should make it clearer in the documentation that replacement names may not contain { or } themselves.

Reply:
Having thought on this some more, I am of the opinion that we should not make this change, because it makes the code more complex (and less efficient, though this is not a particularly a hotspot), only to support an edge case of placeholder keys containing { or } which they realistically never will.

The keys here are all specified by the coders of plugins and never by users, and it would be mad to define a {{xxx}} substitution that contained internal { or } as part of the xxx - it would be a supremely confusing UI.

Also, if we dynamically build the regex, we then need to account for quoting regex metacharacters in key names - the `.` character could plausibly appear in a future key, for example.

I do, though, see that I should document the limitation of not using { } within substitution keys, and add an example of it to the tests.

Would that be an acceptable way to proceed?